### PR TITLE
Add Jenkins job to build a basic Debian image

### PR DESCRIPTION
**Pull request by ana**
Jenkinsfile_basic defines a Jenkins Pipeline creating a stripped-down Debian image for x86, x86_64, arm and arm64. Jenkinsfile_basic contains only the parameters and the code can be found in shared library below var/

The image is created using debos and all the necessary files under the directory of the same name.

Dockerfile_debos provides a docker image with the last version of debos installed.

More details can be found at the discussion starting at https://groups.io/g/kernelci/message/1

### CODE REVIEW

File: vars/buildImage.groovy [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+def call(Closure context) {
+
+    def config = [:]
+    context.resolveStrategy = Closure.DELEGATE_FIRST
+    context.delegate = config
+    context()
+
+    def name = config.name
+    def kernel_arch = config.archList
+    def debian_arch =  ["arm": "armhf",
```
Comment (khilman) [inline vars/buildImage.groovy]: I think we need armel and armhf.  kernelCI still has several ARMv5 boards that will need armel.

File: jenkins/debian/debos/stretch.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  - action: run
+    description: Install extra packages
+    chroot: true
+    command: apt-get update && apt-get -y install --no-install-recommends {{ $extra_packages }}
+
+  - action: run
+    description: Set hostname
+    chroot: true
+    command: echo debian > /etc/hostname
```
Comment (khilman) [inline jenkins/debian/debos/stretch.yaml]: Could you add $architecture to the hostname?  Ideally even the debian distro name.  e.g. "arm64-debian-stretch", or just "arm64-stretch".

File: jenkins/debian/debos/stretch.yaml [hunk 3]
```diff
@@ -68,6 +70,15 @@ actions:
...
     command: rm /usr/lib/tmpfiles.d/dbus.conf
 
+  - action: run
+    description: update-initramfs step
+    chroot: true
+    script: scripts/create_tiny_ramdisk.sh
+
+  - action: run
+    chroot: false
+    command: mv ${ROOTDIR}/boot/initrd.img-min ${ARTIFACTDIR}/{{ $basename -}}/tiny.rootfs.cpio.gz
```
Comment (khilman) [inline jenkins/debian/debos/stretch.yaml]: minor nit: I propose we s/tiny.rootfs/initrd/.  IOW, mv initrd.img-min -> initrd.cpio.gz

File: jenkins/debian/debos/stretch.yaml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  - action: run
+    description: Set symbolic link to init
+    chroot: true
+    command: ln -s /usr/bin/systemd /init
+
+  - action: run
+    description: Create cpio archive
+    chroot: false
+    command: cd ${ROOTDIR} ; find -H  |  cpio -H newc -v -o | gzip -c - > ${ARTIFACTDIR}/{{ $basename -}}/rootfs.cpio.gz
+
```
Comment (khilman) [inline jenkins/debian/debos/stretch.yaml]: Can you add another step to create a compressed ext[234] rootfs?

**Comment by ana** [comment]
I have pushed the commits rebased with some changes:
- there are both arm 32 bits arches now: armel and armhf
- the hostname of the images is debian-$codename-$arch
- renamed tiny.rootfs to initrd.cpio.gz

There is a new commit adding 3 new images:
- a chroot in a ext4 partition: rootfs-${arch}.ext4.xz
- full.rootfs-${arch}.tar.xz
- full.rootfs-${arch}.cpio.gz


I'm afraid that github has been sending a mail every time I made a change.
Please ignore those and check directly the 4 commits above and sorry for the noise!

**Comment by ana** [comment]
I just added a new commit uploading the images. 
A quick note about two things that must be changed if accepting this pull request:
- Jenkins credentials must be updated to match the credentials used by kernelci at: https://github.com/kernelci/kernelci-build-staging/pull/25/files#diff-5bfc323bc0acf7dc2d03ef42943c2a9dR68
- Jenkinsfile_basic must point to the kernelci repo and the master branch and not to my repo and branch:
https://github.com/kernelci/kernelci-build-staging/pull/25/files#diff-1d37183ce146a07e0f9a5c9dccf3b3b4R1
and https://github.com/kernelci/kernelci-build-staging/pull/25/files#diff-1d37183ce146a07e0f9a5c9dccf3b3b4R3

**Comment by khilman** [comment]
Merged this into master, and will add the couple extra commits to switch from using shared libs in Ana's repo to the staging repo.

**Pull request merged**